robot/impl/resource_manager_test.go

```diff
@@ -1838,13 +1838,18 @@ func TestReconfigureParity(t *testing.T) {
 		"data/diff_config_deps11.json",
 		"data/diff_config_deps12.json",
 	}
-	logger := logging.NewTestLogger(t)
 	ctx := context.Background()
 
 	testReconfigureParity := func(t *testing.T, initCfg, updateCfg string) {
 		name := fmt.Sprintf("%s -> %s", initCfg, updateCfg)
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
+			// Capture logs for this sub-test run. Only output the logs if the test fails.
+			logger := logging.NewInMemoryLogger(t)
+			logOnFailure := rutils.NewGuard(func() {
+				logger.OutputLogs()
+			})
+			defer logOnFailure.OnFail()
```

Comment:
neat, didn't know about this guard util. out of curiosity, would a "vanilla" defer that checks the status of the test also work or is there a subtle bug?

e.g. 
```suggestion
			defer func() {
			        if t.Failed() {
				        logger.OutputLogs()
				}
			}
```

Reply:
Oh yeah, that should work. Didn't realize there was a `Testing.Failed` method!